### ISSUE TITLE
Compiler crash with NPE for `[(int|byte)[]...] t4 = [[17], [23, 700]];` at statement level

### ISSUE THREAD

**Issue by lochana-chathura** [ballerina-platform/ballerina-lang#33146]
**Description:**
Consider the following valid code.
```bal
public function main() {
    [(int|byte)[]...] t4 = [[17], [23, 700]]; // compiler crash for this line
    io:println(typeof t4); // @output typedesc [(int|byte)[]...]
    
    [int, (int|byte)[]...] t5 = [12, [17], [23, 700]]; // this line works as expected
    io:println(typeof t5); // @output typedesc [int,(int|byte)[]...]
}

[(int|byte)[]...] t6 = [[17], [23, 700]]; // works at top-level
```

<details>
  <summary><b>stacktrace</b>  (<i>click to expand</i>)</summary>

```
[2021-10-11 11:30:27,923] SEVERE {b7a.log.crash} - null 
java.util.NoSuchElementException
[2021-10-11 11:32:53,047] SEVERE {b7a.log.crash} - null 
java.util.NoSuchElementException
        at java.base/java.util.ArrayDeque.removeFirst(ArrayDeque.java:363)
        at java.base/java.util.ArrayDeque.pop(ArrayDeque.java:594)
        at io.ballerina.compiler.internal.parser.AbstractParserErrorHandler.endContext(AbstractParserErrorHandler.java:235)
        at io.ballerina.compiler.internal.parser.AbstractParser.endContext(AbstractParser.java:127)
        at io.ballerina.compiler.internal.parser.BallerinaParser.parseCompUnit(BallerinaParser.java:180)
        at io.ballerina.compiler.internal.parser.BallerinaParser.parse(BallerinaParser.java:99)
        at io.ballerina.compiler.syntax.tree.SyntaxTree.from(SyntaxTree.java:55)
        at io.ballerina.projects.DocumentContext.parse(DocumentContext.java:88)
        at io.ballerina.projects.DocumentContext.syntaxTree(DocumentContext.java:92)
        at io.ballerina.projects.DocumentContext.getModuleLoadRequests(DocumentContext.java:129)
        at io.ballerina.projects.DocumentContext.moduleLoadRequests(DocumentContext.java:123)
        at io.ballerina.projects.ModuleContext.populateModuleLoadRequests(ModuleContext.java:178)
        at io.ballerina.projects.PackageResolution.getModuleLoadRequestsOfDirectDependencies(PackageResolution.java:161)
        at io.ballerina.projects.PackageResolution.createDependencyGraphFromSources(PackageResolution.java:238)
        at io.ballerina.projects.PackageResolution.buildDependencyGraph(PackageResolution.java:147)
        at io.ballerina.projects.PackageResolution.<init>(PackageResolution.java:71)
        at io.ballerina.projects.PackageResolution.from(PackageResolution.java:78)
        at io.ballerina.projects.PackageContext.getResolution(PackageContext.java:213)
        at io.ballerina.projects.PackageCompilation.<init>(PackageCompilation.java:67)
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:92)
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:206)
        at io.ballerina.projects.Package.getCompilation(Package.java:135)
        at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:64)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.BuildCommand.execute(BuildCommand.java:302)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:58)
```
</details>

**Related Issues:** 
#33143

**Affected Versions:**
SLBeta2

**Comment by lochana-chathura** [comment]
The reason for the crash is #33576

**Comment by lochana-chathura** [comment]
Fixed with #33578
